core/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/Source.scala

```diff
@@ -210,14 +215,29 @@ abstract class Source(
         new DateTime(System.currentTimeMillis),
         canonicalInput
       )
-    processedEvents.map {
-      case Valid(ee) =>
-        (
-          tabSeparateEnrichedEvent(ee),
-          getProprertyValue(ee, partitionKey),
-          getPiiEvent(ee).map(tabSeparateEnrichedEvent)
-        ).valid
-      case Invalid(br) => (br, Random.nextInt().toString()).invalid
+    ) match {
+      case Left(throwable) =>
+        log.error(
+          s"Problem occured while processing CollectorPayload [$canonicalInput]",
+          throwable
+        )
+        sentryClient.foreach { client =>
+          client.sendException(throwable)
+        }
+        Nil
+      case Right(processedEvents) =>
```

Comment:
I personally think it would be easier to match it against:

```scala
case Left(throwable) =>???
case Right(Invalid(badRow)) => ???
case Right(Valid(enriched)) => ???
```

But that's minor, you can ignore if time is pressing.

Reply:
Indeed time is pressing and although this is a simple change, we will have no time to test it, so keeping it in mind for later.